### PR TITLE
`MultiprocessParallelUpdater` to support new devices

### PR DESCRIPTION
**Pull request by hvy**
Support new chainer devices with `MultiProcessParallelUpdater`.

Required for https://github.com/chainer/chainer/pull/7164.

### CODE REVIEW

File: tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py [hunk 1]
```diff
@@ -150,18 +150,18 @@ def setUp(self):
...
         assert ret == 0, (
             '[stdout]:{!r}\n'
             '[stderr]:{!r}'.format(stdoutdata, stderrdata))
 
 
 class TestChildReporter(unittest.TestCase):
 
-    def check_with_gpus(self, n_devices):
-        device_ids_str = ','.join([str(n) for n in range(n_devices)])
+    def check_with_devices(self, *devices):
```
Comment (niboshi) [inline tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py]: Why did you change the interface of this method?
The previous one looks better for me.
Basically `*` is a syntactic sugar and does not seem appropriate in this situation.

File: tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py [hunk 2]
```diff
@@ -191,4 +191,21 @@ def test_cuda_init(self):
             '[stderr]:{!r}'.format(stdoutdata, stderrdata))
 
 
+class TestDevicesByDeviceIds(unittest.TestCase):
+
+    def setUp(self):
+        pass
```
Comment (niboshi) [inline tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py]: Remove the empty `setUp`.

File: tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py (hunk 1 above)
Comment (hvy) [reply tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py]: Fixed.

File: tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py (hunk 2 above)
Comment (hvy) [reply tests/chainer_tests/training_tests/updaters_tests/test_multiprocess_parallel_updater.py]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM
